src/bin/flamegraph.rs

```diff
@@ -18,21 +21,21 @@ impl Into<Options> for Opt {
 }
 
 fn main() -> quick_xml::Result<()> {
-    let (infile, options) = {
+    let (infiles, options) = {
         let opt = Opt::from_args();
-        (opt.infile.clone(), opt.into())
+        let infiles = opt
+            .infiles
+            .iter()
+            .map(|f| File::open(f).map_err(quick_xml::Error::Io))
+            .collect::<Result<Vec<_>, _>>()?;
+        (infiles, opt.into())
     };
 
-    match infile {
-        Some(ref f) => {
-            let r =
-                BufReader::with_capacity(128 * 1024, File::open(f).map_err(quick_xml::Error::Io)?);
-            flamegraph::from_reader(options, r, io::stdout().lock())
-        }
-        None => {
-            let stdin = io::stdin();
-            let r = BufReader::with_capacity(128 * 1024, stdin.lock());
-            flamegraph::from_reader(options, r, io::stdout().lock())
-        }
+    if infiles.is_empty() {
+        let stdin = io::stdin();
+        let r = BufReader::with_capacity(128 * 1024, stdin.lock());
+        flamegraph::from_readers(options, iter::once(r), io::stdout().lock())
+    } else {
```

Comment:
Maybe special-case when `infiles.len() == 1`, since we then don't need to re-sort?